fe/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -4825,6 +4833,50 @@ public void replayRenameTable(TableInfo tableInfo) throws DdlException {
         }
     }
 
+    public void modifyViewDef(Database db, View view, String inlineViewDef) throws DdlException {
+        String viewName = view.getName();
+
+        db.dropTable(viewName);
```

Comment:
I think we should make drop table and create view an atomic operation. Otherwise after view is dropped, concurrent query may create another view, which would make this operation fail.

Reply:
> I think we should make drop table and create view an atomic operation. Otherwise after view is dropped, concurrent query may create another view, which would make this operation fail.

It is atomic, cause it hold the db's writelock before calling this method.